### PR TITLE
grep: fix broken links in gnuman directory

### PR DESCRIPTION
**Pull request by mnencia**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

The fgrep and egrep commands have been deprecated since 2007 and the
related manpages have been removed in 2021. See commit
https://git.savannah.gnu.org/cgit/grep.git/commit/?id=a951562
in the grep source repository.

This patch stops creating broken links for fgrep.1 and egrep.1
manpages in the libexec/gnuman/man1 directory.

### CODE REVIEW

File: Formula/g/grep.rb [hunk 1]
```diff
@@ -53,8 +53,8 @@ def install
     if OS.mac?
       %w[grep egrep fgrep].each do |prog|
         (libexec/"gnubin").install_symlink bin/"g#{prog}" => prog
-        (libexec/"gnuman/man1").install_symlink man1/"g#{prog}.1" => "#{prog}.1"
       end
+      (libexec/"gnuman/man1").install_symlink man1/"ggrep.1" => "grep.1"
```
Comment (carlocab) [inline Formula/g/grep.rb:57]: Let's just do this so we don't have to manually keep track of what lives in the `bin` and `man1` directories:
```suggestion
      (libexec/"gnubin").install_symlink bin.children.to_h do |file|
        [file, file.basename.to_s.delete_prefix("g")]
      end
      (libexec/"gnuman/man1").install_symlink man1.children.to_h do |file|
        [file, file.basename.to_s.delete_prefix("g")]
      end
```

Comment (mnencia) [reply Formula/g/grep.rb:57]: Thanks, this is a really good suggestion.

Comment (mnencia) [reply Formula/g/grep.rb]: I have not been able to make your syntax with `.to_h` working, so I ended up with

```ruby
      bin.children.each do |file|
        (libexec/"gnubin").install_symlink file => file.basename.to_s.delete_prefix("g")
      end
      man1.children.each do |file|
        (libexec/"gnuman/man1").install_symlink file => file.basename.to_s.delete_prefix("g")
      end
```

**Comment by mnencia** [comment]
It looks like the tests failed due to a network timeout. Can you try running them again?

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request merged**